### DISCUSSION TITLE
Removing scanned-page coloration

### DISCUSSION THREAD

**Discussion by faxxy01** [ocrmypdf/OCRmyPDF#1383] [Q&A]
Is there a way to use ocrmypdf to remove the coloration of old pages which have been scanned?   --clean doesn't seem to work.
Can ocrmypdf extract the text and insert it into a new pdf without importing the page coloration?

**Comment by jbarlow83** [comment]
Not currently. ocrmypdf used to have `--remove-background` but the library that supported that feature was too problematic and had to be cut. scikit-image has code that would work but this would add several large dependencies to ocrmypdf which I'm reluctant to do since it would be complicated for downstream packagers. Improving this situation is on the roadmap, but I have no timeline.

I'm also quite picky. I really want features to work >99% of the time, and low quality background removal risks removal of things people care about, like faded comments written in pencil. I'd rather not have it, than have a mediocre feature.

`--clean-final` might help a bit.
